oak-doc/src/site/markdown/security/authorization/bestpractices.md

```diff
@@ -0,0 +1,316 @@
+its members have full access.
+
+#### Stick with group principals
+
+It is preferable to setup access control for group principals instead of individual user principals and then make sure 
+your `PrincipalProvider` resolves principal membership according to your needs.
+
+Further note, that the default authorization model will give precedence to user principals upon evaluation in other words 
+default access control entries for user principals will overwrite the effect of groups irrespective of the order in the list (see next section).
+
+### Understand default access control and permission management
+
+#### Remember inheritance
+
+When designing your access control setup keep in mind that effective permissions are inherited
+down the node hierarchy: allowing `jcr:read` for _content-readers_ role on /content will also grant _content-readers_ 
+access to all nodes and properties in the subtree (e.g. /content/project1 or /content/project1/jcr:title).
+
+In addition, effective permissions get inherited through (nested) group principals according to the set of 
+principals resolved and added to the `javax.security.auth.Subject` upon repository login.
+
+See [Permission Evaluation in Detail](../permission/evaluation.html) for additional information as well as the 
+exercises at [L3_PrecedenceRulesTest](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-exercise/src/test/java/org/apache/jackrabbit/oak/exercise/security/authorization/permission/L3_PrecedenceRulesTest.java)
+
+#### Built-in privileges
+
+JSR 382 defines a set of built-in privileges and how they apply to repository operations (see https://s.apache.org/jcr-2.0-javadoc/javax/jcr/security/Privilege.html).
+The default set has been extended by Oak to cover additional features outside of the scope defined by JCR (like e.g. index 
+or user management). The complete list can be found in [Privilege Management : The Default Implementation](../privilege/default.html). 
+
+The minimal set of privileges required for each operation are outlined in [Mapping API Calls to Privileges](../privilege/mappingtoprivileges.html) 
+and [Mapping Privileges to Items](../privilege/mappingtoitems.html). 
+
+##### Privileges affecting the parent node
+
+Note in particular for add/removing a node `jcr:addChildNodes` and `jcr:removeChildNodes` are required on 
+the parent node respectively i.e. allowing for modification of the child-node collection. 
+In addition `jcr:removeNode` needs to be granted on the target node of the removal.
+
+Thus the following subtle difference apply when evaluation effective permissions vs. privileges (see 
+also [Permissions vs Privileges](../permission/permissionsandprivileges.html)) and exercises at 
+[L4_PrivilegesAndPermissionsTest.java](https://github.com/apache/jackrabbit-oak/blob/trunk/oak-exercise/src/test/java/org/apache/jackrabbit/oak/exercise/security/authorization/permission/L4_PrivilegesAndPermissionsTest.java)):
+
+        String parentPath = /content/parent
```

Comment:
I would prefer if this is formatted as valid java code, including quoting and the semicolon.

Reply:
vale!